spec.bs

```diff
@@ -320,26 +421,243 @@ maps to an internal [=fenced frame config=] [=struct=].
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>fenced frame reporter</dfn>
+  :: null, or a struct containing a [=fencedframetype/fenced frame reporter=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>exfiltration budget metadata</dfn>
+  :: null, or a struct containing an [=fencedframetype/exfiltration budget metadata=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>nested configs</dfn>
+  :: null, or a struct containing a sequence of [=fenced frame configs=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>embedder shared storage context</dfn>
+  :: null, or an [=fencedframetype/embedder shared storage context=]
+</dl>
+
+A <dfn export>fenced frame config instance</dfn> is a struct with the following [=struct/items=]:
+
+<dl export dfn-for="fenced frame config instance">
+  : <dfn>mapped url</dfn>
+  :: null, or a struct containing a [=fencedframetype/mapped url=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>container size</dfn>
+  :: null, or a struct containing a [=fencedframetype/size=] and a
+     [=visibility/visibility to content=]
+
+  : <dfn>content size</dfn>
+  :: null, or a struct containing a [=fencedframetype/size=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>interest group descriptor</dfn>
+  :: null, or a struct containing an [=fencedframetype/interest group descriptor=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>effective sandbox flags</dfn>
+  :: null, or a struct containing an [=fencedframetype/exhaustive set of sandbox flags=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>effective permissions</dfn>
+  :: null, or a struct containing an [=fencedframetype/exhaustive set of permissions=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>fenced frame reporter</dfn> TODO: including automatic beacon info
+  :: null, or a struct containing a [=fencedframetype/fenced frame reporter=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
 
+  : <dfn>exfiltration budget metadata reference</dfn>
+  :: null, or a struct containing an [=fencedframetype/exfiltration budget metadata reference=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>nested configs</dfn>
+  :: null, or a struct containing a sequence of pairs of [=urn uuids=] and [=fenced frame configs=]
+     and a [=visibility/visibility to embedder=]
+
+  : <dfn>partition nonce</dfn>
+  :: a [=partition nonce=]
+
+  : <dfn>embedder shared storage context</dfn>
+  :: null, or an [=fencedframetype/embedder shared storage context=]
+</dl>
+
+<div algorithm="config instantiation">
+  A [=fenced frame config instance=] is initialized from a [=fenced frame config=] as follows:
```

Comment:
Let's have this algorithm explicitly return "a new fenced frame config instance, with:" _and then_ list the members below.